persistent/Database/Persist/Class/PersistQuery.hs

```diff
@@ -22,13 +22,13 @@ class (PersistCore backend, PersistStoreRead backend) => PersistQueryRead backen
     -- | Get all records matching the given criterion in the specified order.
     -- Returns also the identifiers.
     selectSourceRes
-           :: (PersistRecordBackend record backend, MonadIO m1, MonadIO m2)
+           :: (PersistEntity record, PersistRecordBackend record backend, MonadIO m1, MonadIO m2)
```

Comment:
Isn't this redundant? `PersistRecordBackend record backend` is a synonym for `PersistEntity Record` and something else iirc